presto-main/src/main/java/io/prestosql/execution/RefreshMaterializedViewTask.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.execution;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import io.prestosql.Session;
+import io.prestosql.connector.CatalogName;
+import io.prestosql.metadata.Metadata;
+import io.prestosql.metadata.QualifiedObjectName;
+import io.prestosql.security.AccessControl;
+import io.prestosql.sql.analyzer.FeaturesConfig;
+import io.prestosql.sql.parser.SqlParser;
+import io.prestosql.sql.tree.Expression;
+import io.prestosql.sql.tree.RefreshMaterializedView;
+import io.prestosql.transaction.TransactionManager;
+
+import javax.inject.Inject;
+
+import java.util.List;
+
+import static com.google.common.util.concurrent.Futures.immediateFuture;
+import static io.prestosql.metadata.MetadataUtil.createQualifiedObjectName;
+import static java.util.Objects.requireNonNull;
+
+public class RefreshMaterializedViewTask
+        implements DataDefinitionTask<RefreshMaterializedView>
+{
+    private final SqlParser sqlParser;
+
+    @Inject
+    public RefreshMaterializedViewTask(SqlParser sqlParser, FeaturesConfig featuresConfig)
+    {
+        this.sqlParser = requireNonNull(sqlParser, "sqlParser is null");
+        requireNonNull(featuresConfig, "featuresConfig is null");
+    }
+
+    @Override
+    public String getName()
+    {
+        return "REFRESH MATERIALIZED VIEW";
+    }
+
+    @Override
+    public String explain(RefreshMaterializedView statement, List<Expression> parameters)
+    {
+        return "REFRESH MATERIALIZED VIEW " + statement.getName();
+    }
+
+    @Override
+    public ListenableFuture<?> execute(RefreshMaterializedView statement, TransactionManager transactionManager, Metadata metadata, AccessControl accessControl, QueryStateMachine stateMachine, List<Expression> parameters)
+    {
+        Session session = stateMachine.getSession();
+        QualifiedObjectName name = createQualifiedObjectName(session, statement, statement.getName());
+
+        accessControl.checkCanDeleteFromTable(session.toSecurityContext(), name);
+        accessControl.checkCanInsertIntoTable(session.toSecurityContext(), name);
+
+        // OPEN QUESTION
```

Comment:
Refresh design is outlined here: https://docs.google.com/document/d/1GYIyEhJQ3ngvOPmJYWU-8zwDQRCMXkBDZaSn3c3MF0k/edit# , please ignore the open question here.